test: Yk/benchmarking scripts (#1166)
aneoconsulting/armonik: benchmarking/tests/README.md

# HOW TO USE IT

This document describes how to use the benchmarking scripts.

> Those tests are an example of benchmarking tests using bench and htcmock to measure the performances of ArmoniK.  

Reviewer: ```suggestion
These tests are examples of benchmarking based on  Bench and HtcMock tests to measure the performances of ArmoniK.  
```

> We have to deploy ArmoniK on aws using the parameters file in aws-benchmark folder to deploy ArmoniK with two partitions (bench and htcmock) with 100 pods or 1000 pods for each partition.

Reviewer: ```suggestion
We have to deploy ArmoniK on AWS cloud using the parameters files saved in folder [aws-benchmark](https://github.com/aneoconsulting/ArmoniK/tree/main/benchmarking_scripts/infrastructure/quick-deploy/aws-benchmark) to deploy ArmoniK with two partitions (bench and htcmock) with 100 pods or 1000 pods for each partition.
```

```
To deploy ArmoniK on AWS cloud with 1000 pods of compute plane, from the root of the repository : 
```bash
cd infrastructure/quick-deploy/aws/all  
make deploy PARAMETERS_FILE=<Path to ArmoniK>/infrastructure/quick-deploy/aws-benchmark/parameters_1000pods.tfvars
```
# Global view test scripts tree

From the root of the repository, the scripts of bench and htcmock tests are in [benchmarking/tests](https://github.com/aneoconsulting/ArmoniK/tree/main/benchmarking/tests) as follows : 

<pre>
.
├── bench
│   ├── python-scripts
│   │   ├── cleaner.py
│   │   ├── merge-jsons.py
│   │   └── wjson.py
│   ├── stats
│   │   └── test-env.json
│   └── test_scripts
│       ├── bench-100k.sh
│       ├── bench-10k.sh
│       ├── bench-1k.sh
│       ├── bench-5k.sh
│       ├── htcmock-1m.sh
│       └── test.sh
├── htcmock
│   ├── python-scripts
│   │   ├── cleaner.py
│   │   └── merge-jsons.py
│   ├── stats
│   │   └── test-env.json
│   └── test-scripts
│       ├── htcmock-1k.sh
│       ├── htcmock-5k.sh
│       ├── htcmock-10k.sh
│       ├── htcmock-100k.sh
│       ├── htcmock-1m.sh
│       └── test.sh
└── README.md
</pre>

# Bench & HtcMock

## Run the tests

> * `bench-1k.sh`, `bench-5k.sh`, `bench-10k.sh`, `bench-100k.sh` and  `bench-1m.sh`: bash scripts to launch bench tests with 1000 tasks, 5000 tasks, 10000 tasks and 100000 tasks, respectively. In addition, you can modify in the scripts other parameters like workload time, io size, ...

Reviewer: ```suggestion
* `bench-1k.sh`, `bench-5k.sh`, `bench-10k.sh`, `bench-100k.sh` and  `bench-1m.sh`: bash scripts to launch bench tests with 1000 tasks, 5000 tasks, 10000 tasks, 100000 tasks and 1000000 tasks, respectively. In addition, you can modify in the scripts other parameters like workload time, io size, ...
```

> * * `bench-1k.sh`, `bench-5k.sh`, `bench-10k.sh`, `bench-100k.sh` and  `bench-1m.sh`: bash scripts to launch bench tests with 1000 tasks, 5000 tasks, 10000 tasks, 100000 tasks and 1000000 tasks, respectively. In addition, you can modify in the scripts other parameters like workload time, io size, ...

Reviewer: ```suggestion
* `bench-1k.sh`, `bench-5k.sh`, `bench-10k.sh`, `bench-100k.sh` and  `bench-1m.sh`: bash scripts to launch bench tests with 1000 tasks, 5000 tasks, 10000 tasks, 100000 tasks and 1000000 tasks, respectively. In addition, you can modify in the scripts other parameters like workload time, io size, ...
* `htcmock-1k.sh`, `htcmock-5k.sh`, `htcmock-10k.sh`, `htcmock-100k.sh` and  `htcmock-1m.sh`: bash scripts to launch htcmock tests with 1000 tasks, 5000 tasks, 10000 tasks, 100000 tasks and 1000000 tasks, respectively. In addition, you can modify in the scripts other parameters like workload time, io size, ...
```

* `test.sh` : script to run all the tests cases listed above and store the results in JSON files.

### Clean the output

> * cleaner.py : will clean the json output files.  

Reviewer: ```suggestion
* `cleaner.py` : Python script to clean the JSON output files.  
```
